Fix rflink tests broken by Python 3.14.3 asyncio changes

### Description

The following `rflink` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/rflink/test_binary_sensor.py::test_entity_availability` (failed 6/10)**

```
tests/components/rflink/test_binary_sensor.py:106: in test_entity_availability
    assert hass.states.get("binary_sensor.test").state == STATE_UNKNOWN
E   AssertionError: assert 'unavailable' == 'unknown'
E     
E     - unknown
E     + unavailable
```

**`tests/components/rflink/test_sensor.py::test_entity_availability` (failed 5/10)**

```
tests/components/rflink/test_binary_sensor.py:106: in test_entity_availability
    assert hass.states.get("binary_sensor.test").state == STATE_UNKNOWN
E   AssertionError: assert 'unavailable' == 'unknown'
E     
E     - unknown
E     + unavailable
tests/components/rflink/test_sensor.py:150: in test_entity_availability
    assert hass.states.get("sensor.test").state == STATE_UNKNOWN
E   AssertionError: assert 'unavailable' == 'unknown'
E     
E     - unknown
E     + unavailable
```


#### What was done

These tests have been temporarily disabled with `@pytest.mark.xfail` in PR #169074.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169074

--- a/tests/components/rflink/test_binary_sensor.py
+++ b/tests/components/rflink/test_binary_sensor.py
@@ -97,13 +97,14 @@ async def test_entity_availability(
     # Make sure Rflink mock does not 'recover' to quickly from the
     # disconnect or else the unavailability cannot be measured
     config = CONFIG
-    failures = [True, True]
+    failures = [False, True]
     config[CONF_RECONNECT_INTERVAL] = 60
 
     # Create platform and entities
     event_callback, _, _, disconnect_callback = await mock_rflink(
         hass, config, DOMAIN, monkeypatch, failures=failures
     )
+    await hass.async_block_till_done()
 
     # Entities are unknown by default
     assert hass.states.get("binary_sensor.test").state == STATE_UNKNOWN

--- a/tests/components/rflink/test_binary_sensor.py
+++ b/tests/components/rflink/test_binary_sensor.py
@@ -87,18 +87,20 @@ async def test_default_setup(
     assert hass.states.get("binary_sensor.test").state == STATE_OFF
 
 
-@pytest.mark.xfail(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_entity_availability(
     hass: HomeAssistant, monkeypatch: pytest.MonkeyPatch
 ) -> None:
     """If Rflink device is disconnected, entities should become unavailable."""
     # Make sure Rflink mock does not 'recover' to quickly from the
     # disconnect or else the unavailability cannot be measured
-    config = CONFIG
+    config = {
+        **CONFIG,
+        "rflink": {
+            **CONFIG["rflink"],
+            CONF_RECONNECT_INTERVAL: 60,
+        },
+    }
     failures = [False, True]
-    config[CONF_RECONNECT_INTERVAL] = 60
 
     # Create platform and entities
     event_callback, _, _, disconnect_callback = await mock_rflink(

--- a/tests/components/rflink/test_init.py
+++ b/tests/components/rflink/test_init.py
@@ -58,7 +58,7 @@ async def mock_rflink(
         # failures can be a list of booleans indicating in which sequence
         # creating a connection should success or fail
         if failures:
-            fail = failures.pop()
+            fail = failures.pop(0)  # removes from left to right
         else:
             fail = False
 

--- a/tests/components/rflink/test_sensor.py
+++ b/tests/components/rflink/test_sensor.py
@@ -141,13 +141,14 @@ async def test_entity_availability(
     # Make sure Rflink mock does not 'recover' to quickly from the
     # disconnect or else the unavailability cannot be measured
     config = CONFIG
-    failures = [True, True]
+    failures = [False, True]
     config[CONF_RECONNECT_INTERVAL] = 60
 
     # Create platform and entities
     _, _, _, disconnect_callback = await mock_rflink(
         hass, config, DOMAIN, monkeypatch, failures=failures
     )
+    await hass.async_block_till_done()
 
     # Entities are available by default
     assert hass.states.get("sensor.test").state == STATE_UNKNOWN

--- a/tests/components/rflink/test_sensor.py
+++ b/tests/components/rflink/test_sensor.py
@@ -131,18 +131,20 @@ async def test_disable_automatic_add(
     assert not hass.states.get("sensor.test2")
 
 
-@pytest.mark.xfail(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_entity_availability(
     hass: HomeAssistant, monkeypatch: pytest.MonkeyPatch
 ) -> None:
     """If Rflink device is disconnected, entities should become unavailable."""
     # Make sure Rflink mock does not 'recover' to quickly from the
     # disconnect or else the unavailability cannot be measured
-    config = CONFIG
+    config = {
+        **CONFIG,
+        "rflink": {
+            **CONFIG["rflink"],
+            CONF_RECONNECT_INTERVAL: 60,
+        },
+    }
     failures = [False, True]
-    config[CONF_RECONNECT_INTERVAL] = 60
 
     # Create platform and entities
     _, _, _, disconnect_callback = await mock_rflink(

